psalm-baseline.xml

```diff
@@ -1,3 +1,129 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<files psalm-version="5.15.0@5c774aca4746caf3d239d9c8cadb9f882ca29352">
+<files psalm-version="5.20.0@3f284e96c9d9be6fe6b15c79416e1d1903dcfef4">
+  <file src="src/Builder/Encoder/AbstractExpressionEncoder.php">
+    <MixedAssignment>
+      <code>$val</code>
+      <code>$val</code>
+      <code>$value[$key]</code>
+    </MixedAssignment>
+  </file>
+  <file src="src/Builder/Encoder/CombinedFieldQueryEncoder.php">
+    <MixedAssignment>
+      <code>$filter</code>
+      <code>$filterValue</code>
+    </MixedAssignment>
+  </file>
+  <file src="src/Builder/Encoder/FieldPathEncoder.php">
+    <NoInterfaceProperties>
+      <code><![CDATA[$value->name]]></code>
+    </NoInterfaceProperties>
+  </file>
+  <file src="src/Builder/Encoder/OperatorEncoder.php">
+    <MixedAssignment>
+      <code>$result</code>
+      <code>$result[]</code>
+      <code>$val</code>
+      <code>$val</code>
+      <code>$val</code>
+      <code>$val</code>
+      <code>$val</code>
+    </MixedAssignment>
+    <UndefinedConstant>
+      <code>$value::ENCODE</code>
+    </UndefinedConstant>
+  </file>
+  <file src="src/Builder/Encoder/QueryEncoder.php">
+    <MixedAssignment>
+      <code>$subValue</code>
+      <code>$value</code>
+    </MixedAssignment>
+  </file>
+  <file src="src/Builder/Projection/ElemMatchOperator.php">
+    <MixedArgumentTypeCoercion>
```

Comment:
`MixedArgumentTypeCoercion` errors are due to the query expression not using the correct array element types but instead using the `array` type. I tried to fix this, but the code generator doesn't accept a type like `array<int|string>`. This will have to be looked at separately.